src/Workleap.Extensions.MediatR.Analyzers/SemanticDesignAnalyzer.cs

```diff
@@ -101,15 +117,30 @@ public void AnalyzeNamedType(SymbolAnalysisContext context)
 
         private bool ImplementsHandlerInterface(ITypeSymbol type)
         {
-            return type.AllInterfaces.Any(this.IsHandler);
+            return this.ImplementsRequestHandlerInterface(type) || this.ImplementsNotificationHandlerInterface(type);
+        }
+
+        private bool ImplementsRequestHandlerInterface(ITypeSymbol type)
+        {
+            return type.AllInterfaces.Any(this.IsRequestHandler);
+        }
+
+        private bool IsRequestHandler(INamedTypeSymbol symbol)
+        {
+            return symbol is { IsGenericType: true, Arity: 1 or 2 } && this._requestHandlerTypes.Contains(symbol.ConstructedFrom);
+        }
+
+        private bool ImplementsNotificationHandlerInterface(ITypeSymbol type)
+        {
+            return type.AllInterfaces.Any(this.IsNotificationHandler);
         }
 
-        private bool IsHandler(INamedTypeSymbol symbol)
+        private bool IsNotificationHandler(INamedTypeSymbol symbol)
         {
-            return symbol is { IsGenericType: true, Arity: 1 or 2 } && this._handlerTypes.Contains(symbol.ConstructedFrom);
+            return symbol is { IsGenericType: true, Arity: 1 or 2 } && this._notificationHandlerTypes.Contains(symbol.ConstructedFrom);
         }
```

Comment:
My preference, but I'd regroup those two in just one

Reply:
I think this is more readable and results in smaller methods without a large one-liner logic statement.